### DISCUSSION TITLE
Development of quanteda v4.0

### DISCUSSION THREAD

**Discussion by koheiw** [quanteda/quanteda#2209] [General]
Next major release of quanteda will be v4.0. It would include:

- Introduce new tokens object 
   -  #2208
- Defunct deprecated functions and arguments
- Redesign or defunct some of the unreliable functions (e.g. `tokens_match()`)
- Remove old examples (e.g. quickstart.Rmd)

What else would be the changes in the version?

**Comment by koheiw** [comment]
- Development of new tokenizer
   -  #2165, #2216
- Development of new serilizer
   - #2218

**Comment by kbenoit** [comment]
So @koheiw I'm ready to begin review of the PRs and starting actively talking about 4.0. 

What's the sequence of PR reviews? If I have understood correctly, it's
1.  #2216
2. #2210 
then...?

**Comment by kbenoit** [comment]
@koheiw I've done all of the PR reviews now except #2210, but I see that we have two parallel branches, and I don't want to get them drifted too far apart. The other three pending PRs should be merged to master, which i would have done, but wanted to confirm with you first that this is alright.

So:
* Any fix that does not depend on new v4 changes, we merge to both `master` and `v4`
* Any fix that is v4 specific, we merge to just `v4`. This will require freezing v3 at some stage, but I think there are some small issues we could still address that would become part of a last v3.x version. That means we are not pressured to release v4 because of bug fixes or improvements that belong in v3 but that we made only to the increasingly different v4 branch.

**Comment by kbenoit** [comment]
Following @koheiw's [comment](https://github.com/quanteda/quanteda/pull/2210#issuecomment-1493684259) about the xtokens branch, let's keep a parallel branch v4, but try to keep that branch as up-to-date as possible with v3 bug fixes and deprecations.

I still think we could fix the following issues for v3:
- #2218 - but @koheiw to confirm
- #2103 
- #2105
